pkg/operator/controllers/checker/unsupportconditionchecker.go

```diff
@@ -0,0 +1,122 @@
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	clusterOperatorName = "aro"
+	reasonAsExpected    = "AsExpected"
+)
+
+type NodesChecker struct {
+	kubernetescli kubernetes.Interface
+	consolecli    consoleclient.Interface
+	configcli     configclient.Interface
+}
+
+func NewNodesChecker(kubernetescli kubernetes.Interface, consolecli consoleclient.Interface, configcli configclient.Interface) *NodesChecker {
+	return &NodesChecker{
+		kubernetescli: kubernetescli,
+		consolecli:    consolecli,
+		configcli:     configcli,
+	}
+}
+
+func (nc *NodesChecker) Name() string {
+	return "WorkerNodesCountChecker"
+}
+
+func (nc *NodesChecker) Check(ctx context.Context) error {
+	bannerName := "aro-sre"
+	nodes, err := nc.kubernetescli.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+		LabelSelector: "node-role.kubernetes.io/worker",
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(nodes.Items) < 3 {
+		// Set the Cluster Operator Upgradable status to "False"
+		nc.setCluOperatorStatus(ctx, 0)
+
+		// Call the newBanner function to display notification on the banner
+		banner := nc.newBanner(bannerName, "The number of the current worker nodes are less then 3")
+
+		_, err := nc.consolecli.ConsoleV1().ConsoleNotifications().Get(ctx, bannerName, metav1.GetOptions{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return err
+		}
+
+		// if the object doesn't exist Create
+		if err != nil && kerrors.IsNotFound(err) {
+			_, err = nc.consolecli.ConsoleV1().ConsoleNotifications().Create(ctx, banner, metav1.CreateOptions{})
+			return err
+		}
+
+		// if there's no errors, object found then update
+		_, err = nc.consolecli.ConsoleV1().ConsoleNotifications().Update(ctx, banner, metav1.UpdateOptions{})
+		return err
+	} else {
+		nc.setCluOperatorStatus(ctx, 1)
+
+		err = nc.consolecli.ConsoleV1().ConsoleNotifications().Delete(ctx, bannerName, metav1.DeleteOptions{})
+		return err
+	}
+}
+
+func (nc *NodesChecker) setCluOperatorStatus(ctx context.Context, status_index int) {
+	co, _ := nc.getOrCreateClusterOperator(ctx)
+
+	currentTime := metav1.Now()
+	condition := []configv1.ClusterOperatorStatusCondition{
+		{
+			Type:               configv1.OperatorUpgradeable,
+			Status:             configv1.ConditionFalse,
+			LastTransitionTime: currentTime,
+			Reason:             reasonAsExpected,
+		},
+		{
+			Type:               configv1.OperatorUpgradeable,
+			Status:             configv1.ConditionTrue,
+			LastTransitionTime: currentTime,
+			Reason:             reasonAsExpected,
+		},
+	}
+	v1helpers.SetStatusCondition(&co.Status.Conditions, condition[status_index])
+}
+
+func (nc *NodesChecker) getOrCreateClusterOperator(ctx context.Context) (*configv1.ClusterOperator, error) {
+	co, err := nc.configcli.ConfigV1().ClusterOperators().Get(ctx, clusterOperatorName, metav1.GetOptions{})
+	if !kerrors.IsNotFound(err) {
```

Comment:
This double negative is really hard to read at first and the function is missing the create step. In fact it should only get the cluster operator and return an error if it can't. The creation is l;eft to the operator's reconcile operation:https://github.com/Azure/ARO-RP/blob/9f472ae57106c793ec1af373627eb77eac86d215/pkg/operator/controllers/clusteroperatoraro/clusteroperatoraro_controller.go#L134-L150